sdk/go/pulumi/context.go

```diff
@@ -677,6 +693,31 @@ func (ctx *Context) prepareResourceInputs(props Input, t string,
 	}
 	sort.Strings(deps)
 
+	// resolve aliases and implicit parent aliases, resolving all to URNs
+	var aliases []string
+	if opts.Parent != nil {
+		pAliases := opts.Parent.getAliases()
+		for _, alias := range pAliases {
+			childAlias, err := inheritedChildAlias(resource.name, opts.Parent.getName(), t, ctx.Project(), ctx.Stack(), alias)
+			if err != nil {
+				return nil, errors.New("Could not convert parent alias to child alias")
+			}
+			resource.aliases = append(resource.aliases, Alias{URN: childAlias})
+		}
+	}
+	for _, alias := range resource.aliases {
+		err = alias.collapseToURN(resource.name, t, opts.Parent, ctx.Project(), ctx.Stack())
+		if err != nil {
+			return nil, errors.Wrap(err, "error collapsing alias to URN")
+		}
+		urnOut := alias.URN.ToURNOutput()
+		urn, _, err := urnOut.awaitURN(context.Background())
```

Comment:
I don't think we want to do any `await`s here.  Can't our `aliases` be an array of `Outputs` as in TypeScript?

Reply:
I may be missing something, but this seems to indicate that all aliases are converted to strings by the time we make the register resource call: 

https://github.com/pulumi/pulumi/blob/972fc44dd78c58da5887e65185ee2f91553cc41e/sdk/python/lib/pulumi/runtime/resource.py#L364

https://github.com/pulumi/pulumi/blob/9abcca345a6c024087b35152b0d6fca92baef191/sdk/nodejs/runtime/resource.ts#L193